network/p2p/middleware.go

```diff
@@ -524,18 +568,32 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 		m.wg.Add(1)
 		go func(msg *message.Message) {
 			defer m.wg.Done()
-			if decodedMsgPayload, err := m.codec.Decode(msg.Payload); err != nil {
+			decodedMsgPayload, err := m.codec.Decode(msg.Payload)
+			if err != nil {
 				m.log.
```

Comment:
this could be slashable. Since the peer was authenticated and the message envelope was verified, the peer actually sent an invalid/unexpected payload. `OnUnknownMsgTypeError`?

Reply:
https://github.com/onflow/flow-go/pull/2832/commits/0b9f1172c4d341fbe2dbd8ec18fe514eeab80511